packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,130 @@
 import { ApiRef } from '../../models';
 
 const MIN_COL_WIDTH = 50;
-const MOUSE_LEFT_TIMEOUT = 1000;
-
-// TODO improve experience for last column
-export const useColumnResize = (
-  columnsRef: React.RefObject<HTMLDivElement>,
-  apiRef: ApiRef,
-  headerHeight: number,
-) => {
-  const logger = useLogger('useColumnResize');
 
-  const isResizing = React.useRef<boolean>(false);
-  const isLastColumn = React.useRef<boolean>(false);
-  const mouseLeftTimeout = React.useRef<any>();
-  const stopResizeEventTimeout = React.useRef<any>();
+export const useColumnResize = (columnsRef: React.RefObject<HTMLDivElement>, apiRef: ApiRef) => {
+  const logger = useLogger('useColumnResize');
+  const colDefRef = React.useRef<ColDef>();
+  const colElementRef = React.useRef<HTMLDivElement>();
+  const colCellElementsRef = React.useRef<NodeListOf<Element>>();
+  const initialOffset = React.useRef<number>();
+  const stopResizeEventTimeout = React.useRef<number>();
+
+  const updateWidth = (newWidth: number) => {
+    logger.debug(`Updating width to ${newWidth} for col ${colDefRef.current!.field}`);
+
+    colDefRef.current!.width = newWidth;
+
+    colElementRef.current!.style.width = `${newWidth}px`;
+    colElementRef.current!.style.minWidth = `${newWidth}px`;
+    colElementRef.current!.style.maxWidth = `${newWidth}px`;
+
+    colCellElementsRef.current!.forEach((element) => {
+      const div = element as HTMLDivElement;
+      div.style.width = `${newWidth}px`;
+      div.style.minWidth = `${newWidth}px`;
+      div.style.maxWidth = `${newWidth}px`;
+    });
+  };
+
+  const handleResizeMouseMove = useEventCallback((nativeEvent) => {
+    // Cancel move in case some other element consumed a mouseup event and it was not fired.
+    if (nativeEvent.buttons === 0) {
+      // @ts-expect-error fixed in v5
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      handleResizeMouseUp(nativeEvent);
+      return;
+    }
 
-  const currentColDefRef = React.useRef<ColDef>();
-  const currentColElem = React.useRef<HTMLDivElement>();
-  const currentColPosition = React.useRef<number>();
-  const currentColPreviousWidth = React.useRef<number>();
-  const currentColCellsElems = React.useRef<NodeListOf<Element>>();
+    let newWidth =
+      initialOffset.current +
+      nativeEvent.clientX -
+      colElementRef.current!.getBoundingClientRect().left;
+    newWidth = Math.max(MIN_COL_WIDTH, newWidth);
 
-  const dataContainerElemRef = React.useRef<HTMLDivElement>();
-  const dataContainerPreviousWidth = React.useRef<number>();
-  const scrollOffset = React.useRef<number>(0);
-  const resizingMouseMove = React.useRef<{ x: number; y: number }>();
+    updateWidth(newWidth);
+  });
 
-  const onScrollHandler = React.useCallback((params: ScrollParams) => {
-    scrollOffset.current = params.left;
-  }, []);
+  const handleResizeMouseUp = useEventCallback(() => {
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
```

Comment:
Same here

Reply:
I fear we have a cycle we can't break so I can't disable the rule with this pattern, I'm also not aware of an alternative pattern. I think that we should keep https://eslint.org/docs/rules/no-use-before-define because it makes the code easier to process (by giving is a structure).